lib/arclight/repository.rb

```diff
@@ -7,31 +7,14 @@ module Arclight
   class Repository
     include ActiveModel::Conversion # for to_partial_path
 
-    FIELDS = %i[name
-                description
-                visit_note
-                building
-                address1
-                address2
-                city
-                state
-                zip
-                country
-                phone
-                contact_info
-                thumbnail_url
-                google_request_url
-                google_request_mappings
-                collection_count].freeze
-
-    attr_accessor :slug, *FIELDS
+    attr_accessor :slug, :collection_count
 
     # @param [String] `slug` the unique identifier for the repository
     # @param [Hash] `data`
     def initialize(slug, data = {})
       @slug = slug
-      FIELDS.each do |field|
-        value = data[field.to_s]
+      data.each do |field, value|
+        self.class.attr_accessor field.to_sym
```

Comment:
There is some potential for future race condition issues here that I don't believe we're susceptible to now.  Happy to talk through my concerns further, or just ship it and deal w it if we ever cross that bridge.

Reply:
Could you just briefly describe the future race condition? So at least we have it shared?

Reply:
I guess ArcLight could reference a field that hasn't been defined?

E.g

I only provide two fields in my config. But it calls a third not defined?

Reply:
My concern is if we were to do something like initialize additional repositories at runtime (which we don't have the facility to do currently) there could be an issue where we tried to use an accessor that hasn't been defined yet.  Maybe it'll never happen, and because `initialize` is at the class level so it all just works out.  I've just never seen the `self.class.attr_accessor` used like this in the wild so I'm just like, Ruby: 

![you cray](https://www.civicx.com/attachments/wf_c2-jpg.32062/)

Reply:
🆒